### DISCUSSION TITLE
Handling address state inside transaction context

### DISCUSSION THREAD

**Discussion by benadamsky** [wevm/frog-archived#163] [General]
I'm currently looking to store the returned address that comes from a transaction. However, `address` only exists inside the app.transaction() context and deriveState only seems to exist inside the app.frame() context, so I'm not sure if there's a way to store an address after the transaction is fired.

Context: Being able to store a transaction address statefully would allow us to maintain parity with our in-frame ERC20 approval tx and separately writing to our smart contract through our backend wallet, instead of guessing which wallet was used via neynar verifications.

**Comment by ggomaeng** [comment]
What I do is fetch the sender address using the `transactionId` in the frame context

**Reply by benadamsky** [reply]
Do you use a provider like Alchemy / Infura to do the lookup? How are response times within the frame?

That's an option as well, but my main concern is that reading onchain data can be time-intensive, especially if I have other calls post-transaction. Will have to do some testing and see if that could be an option for the time being.

**Reply by ggomaeng** [reply]
I use viem's publicClient to look up the tx details. Read calls are pretty fast - but if you want stability you could use Alchemy or other options.
